### DISCUSSION TITLE
How to add React (or single page application) routes to pyinstaller?

### DISCUSSION THREAD

**Discussion by lunaeidi** [pyinstaller/pyinstaller#8598] [Help]
I'm building my Python/Flask+React/Vite app into a single executable using pyinstaller. I'm having an issue where I'm unable to access my React routes when running the app from the executable (debian). Getting an error in the Flask server when navigating to the site: GET / HTTP/1.1" 404 - (which applies that it's trying to hit that endpoint on the Flask server, and is not respecting the React routes)

This only happens when running the debian package.

The routes work fine when just running the Flask server from source. I had added this to my flask code to get that to work:

       `      @app.route('/', defaults={'path': '/'})
        @app.route('/<path:path>')
        @app.endpoint("index")
        def serve(path):
            if path != "/" and os.path.exists(app.static_folder + '/' + path):
                return send_from_directory(app.static_folder, path)
            else:
                return send_from_directory(app.static_folder, 'index.html')
`

Is there a flag to add to the pyinstaller build command that's needed to have my React routes work?

Here's my pyinstaller command:
` pyinstaller --add-data "static:static" --onefile src/$app.py --name $app-$VERSION --log-level WARN --distpath $BUILD_DIR/$COMPONENTS_DIR/$app/pyinstaller/dist/ --workpath $BUILD_DIR/$COMPONENTS_DIR/$app/pyinstaller/temp/`

Thanks!

**Comment by rokm** [comment]
What is `app.static_folder` set to?

**Reply by lunaeidi** [reply]
`
app = Flask(__name__, static_folder='../../client/dist') 
`

I copied the contents of the dist folder that was generated from running npm run build (since I didn't know how to refer directly to the client/dist folder in the pyinstall command), into a folder "static" that's a subdirectory of my server folder. Here's my file structure: 

```
-server (folder)
  -static (folder)
  -build.sh
  -src (folder)
    -flask-app.py
-client
  -dist (folder)
```

Thanks!

**Reply by rokm** [reply]
> `app = Flask(__name__, static_folder='../../client/dist') `

You should collect this `client/dist` directory into your frozen application (similarly as you did with the `static` folder).

And then you should refer to it with absolute path, anchored to either `__file__` or `sys._MEIPASS`, as per docs: https://pyinstaller.org/en/stable/runtime-information.html#run-time-information
